injection/sharedmain/main.go

```diff
@@ -146,18 +146,7 @@ func EnableInjectionOrDie(ctx context.Context, cfg *rest.Config) context.Context
 	}
 	ctx = injection.WithConfig(ctx, cfg)
 
-	ctx, informers := injection.Default.SetupInformers(ctx, cfg)
-
-	// Start the injection clients and informers.
-	logging.FromContext(ctx).Info("Starting informers...")
-	go func(ctx context.Context) {
```

Comment:
The whole point of this method is to start the informers here and not need to duplicate that code in the caller. 

Perhaps we can pass a delay config in the inbound context object instead? The goal is to reduce the amount of special code each injection enabled cmd needs to run. `MainWithConfig` is just one simple example.

Another example of integration: https://github.com/knative-sandbox/eventing-rabbitmq/blob/56208970df1d5db5526a12137d54ea16b37fe149/test/e2e/cmd/recorder/main.go#L32

I don't want to have to copy the informer startup code in all the places I have called enable injection.

Reply:
Delaying the start of goroutine might be a little bit tricky. If we use a magic time duration, then it might cause lots of unit test to fail. If we use a signal instead, then we still need to change all the downstream codes to signal start, which is same as calling `StartInformers` directly.

Reply:
Please check this PR again. I think I prefer a single method that returns a callback over two new methods.